hal_st/stm32fxxx/AdcDmaStm.hpp

```diff
@@ -0,0 +1,39 @@
+#ifndef HAL_ADC_DMA_STM_HPP
+#define HAL_ADC_DMA_STM_HPP
+
+#include "hal/interfaces/AnalogToDigitalPin.hpp"
+#include "hal_st/stm32fxxx/AnalogToDigitalPinStm.hpp"
+#include "hal_st/stm32fxxx/DmaStm.hpp"
+#include "hal_st/stm32fxxx/TimerStm.hpp"
+#include "infra/util/AutoResetFunction.hpp"
+#include "infra/util/InterfaceConnector.hpp"
+
+namespace hal
+{
+    class AdcTriggeredByTimerWithDma
+        : public infra::InterfaceConnector<AdcTriggeredByTimerWithDma>
```

Comment:
Why does this need to derive from InterfaceConnector? As far as I can see, the class does not need to be a singleton

Reply:
Correct, removed.